packages/account/src/Components/terms-of-use/terms-of-use.tsx

```diff
@@ -132,12 +137,30 @@ const TermsOfUse = ({
                                                 />
                                             }
                                         />
+                                        <Hr />
+                                        {is_spain_residence && (
+                                            <Field
+                                                component={CheckboxField}
+                                                className='terms-of-use__checkbox'
+                                                name='resident_self_declaration'
+                                                id='resident_self_declaration'
+                                                label={
+                                                    <Localize i18n_default_text='I hereby confirm that my request for opening an account with Deriv Investments (Europe) Ltd is made on my own initiative.' />
+                                                }
+                                                label_line_height='s'
+                                            />
+                                        )}
                                     </div>
                                 </ThemedScrollbars>
                             </Div100vhContainer>
                             <Modal.Footer has_separator is_bypassed={isMobile()}>
                                 <FormSubmitButton
-                                    is_disabled={isSubmitting || !values.agreed_tos || !values.agreed_tnc}
+                                    is_disabled={
+                                        isSubmitting ||
+                                        !values.agreed_tos ||
+                                        !values.agreed_tnc ||
+                                        !(is_spain_residence ? values.resident_self_declaration : true)
```

Comment:
you may ignore this as well.
shall we move this to a variable,
`const has_self_declaration = is_spain_residence ? !!values.resident_self_declaration : true`